temporalio/client.py

```diff
@@ -2830,6 +3097,130 @@ async def report_cancellation(
+    ) -> ReturnType:
+        """Wait for result of the activity.
+
+        Args:
+            rpc_metadata: Headers used on the RPC call. Keys here override
+                client-level RPC metadata keys.
+            rpc_timeout: Optional RPC deadline to set for each RPC call. Note,
+                this is the timeout for each history RPC call not this overall
+                function.
+
+        Returns:
+            The result of the activity.
+
+        Raises:
+            :py:class:`ActivityFailureError`: If the activity completed with a failure.
+        """
+        # Repeatedly issues workflowservice GetActivityResult long-polls.
+        raise NotImplementedError
+
+    async def cancel(
+        self,
+        *,
+        reason: Optional[str] = None,
+        rpc_metadata: Mapping[str, Union[str, bytes]] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> None:
+        """Request cancellation of the activity.
+
+        Requesting cancellation of an activity does not automatically transition the activity to
+        canceled status. If the activity is heartbeating, a :py:class:`exceptions.CancelledError`
+        exception will be raised when receiving the heartbeat response; if the activity allows this
+        exception to bubble out, the activity will transition to canceled status. If the activity it
+        is not heartbeating, this method will have no effect on activity status.
+
+        Args:
+            reason: Reason for the cancellation. Recorded and available via describe.
+            rpc_metadata: Headers used on the RPC call.
+            rpc_timeout: Optional RPC deadline to set for the RPC call.
+        """
+        raise NotImplementedError
+
+    async def terminate(
+        self,
+        *,
+        reason: Optional[str] = None,
+        rpc_metadata: Mapping[str, Union[str, bytes]] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> None:
+        """Terminate the activity execution immediately.
+
+        Termination does not reach the worker and the activity code cannot react to it.
+        A terminated activity may have a running attempt and will be requested to be
+        canceled by the server when it heartbeats.
+
+        Args:
+            reason: Reason for the termination.
+            rpc_metadata: Headers used on the RPC call.
+            rpc_timeout: Optional RPC deadline to set for the RPC call.
+        """
+        raise NotImplementedError
+
+    async def describe(
+        self,
+        *,
+        rpc_metadata: Mapping[str, Union[str, bytes]] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> ActivityExecutionDescription:
+        """Describe the activity execution.
+
+        Args:
+            rpc_metadata: Headers used on the RPC call.
+            rpc_timeout: Optional RPC deadline to set for the RPC call.
+
+        Returns:
+            Activity execution description.
+        """
+        raise NotImplementedError
+
+    # TODO:
```

Comment:
Also TODO: all of the async completion methods.

Reply:
You mean `heartbeat`, `complete`, `fail`, and `report_cancellation`, right? Those are all inherited from `_BaseActivityHandle`.